app/HutchAgent/Services/ActionHandlers/FetchAndExecuteActionHandler.cs

```diff
@@ -86,11 +88,14 @@ public async Task HandleAction(string jobId, object? payload)
         await _status.ReportStatus(jobId, JobStatus.Failure,
           $"The remote Request Crate was not accepted: ${JsonSerializer.Serialize(acceptResult.Errors)}. Please resubmit the job.");
 
-        if(!await _features.IsEnabledAsync(FeatureFlags.RetainFailures))
+        if (!await _features.IsEnabledAsync(FeatureFlags.RetainFailures))
           await _job.Cleanup(job);
         return;
       }
 
+      // throw invalid data if checksums don't match
+      if (!await _bagIt.VerifyChecksums(jobId.BagItPayloadPath())) throw new InvalidDataException();
```

Comment:
@beforan I'm not sure if `jobId.BagItPayloadPath()` the correct path here.

Reply:
Probably not. The `job` db entity should be in scope, so you should be able to use extension methods to get to the right path e.g. `job.WorkingDirectory.JobBagItRoot()`